src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -0,0 +1,208 @@
+            "LocalFinalVariableName",
+            "LocalVariableName",
+            "MagicNumber",
+            "MemberName",
+            "MethodLength",
+            "MethodName",
+            "MethodTypeParameterName",
+            "ModifiedControlVariable",
+            "ModifierOrder",
+            "MultipleStringLiterals",
+            "MutableException",
+            "OperatorWrap",
+            "PackageName",
+            "ParameterAssignment",
+            "ParameterName",
+            "ParameterNumber",
+            "RedundantImport",
+            "RedundantModifier",
+            "ReturnCount",
+            "SeparatorWrap",
+            "SimplifyBooleanExpression",
+            "SimplifyBooleanReturn",
+            "StaticVariableName",
+            "StringLiteralEquality",
+            "SuperClone",
+            "SuperFinalize",
+            "SuppressWarnings",
+            "ThrowsCount",
+            "TypeName",
+            "VisibilityModifier"
+    ));
+
+    private static Set<String> simpleCheckNames;
+    private static Map<String, String> allowedDirectoryAndChecks;
+
+    private Path javaDir;
+    private Path inputDir;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws IOException {
+        simpleCheckNames = CheckUtil.getSimpleNames(CheckUtil.getCheckstyleChecks());
+
+        allowedDirectoryAndChecks = simpleCheckNames
+                .stream()
+                .collect(Collectors.toMap(id -> id.toLowerCase(Locale.ENGLISH), id -> id));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        javaDir = Paths.get("src/it/java/" + getPackageLocation());
+        inputDir = Paths.get(getPath(""));
+    }
+
+    @Override
+    protected String getPackageLocation() {
+        return "org/checkstyle/suppressionxpathfilter";
+    }
+
+    @Override
+    protected String getResourceLocation() {
+        return "it";
+    }
+
+    @Test
+    public void validateJavaFiles() throws Exception {
+        final Pattern pattern = Pattern.compile("^XpathRegression(.+)Test\\.java$");
+        try (DirectoryStream<Path> javaPaths = Files.newDirectoryStream(javaDir)) {
+            for (Path path : javaPaths) {
+                assertTrue(path + " is not a regular file", Files.isRegularFile(path));
+                final String filename = path.toFile().getName();
+                if (filename.startsWith("Abstract")) {
+                    continue;
+                }
+
+                final Matcher matcher = pattern.matcher(filename);
+                assertTrue("Invalid test file: " + filename + ", expected pattern: " + pattern,
+                        matcher.matches());
+
+                final String check = matcher.group(1);
+                assertTrue("Unknown check '" + check + "' in test file: " + filename,
+                        simpleCheckNames.contains(check));
+
+                assertFalse("Check '" + check + "' is now tested. Please update the todo list.",
```

Comment:
what is todo list ?
it is better to reference as `{Classname}.{field}`

Reply:
done